Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector.py

```diff
@@ -0,0 +1,488 @@
+            if isinstance(attachment, dict):
+                attachment.pop("fileName", None)
+
+    return event
+
+
+def fetch_by_event_type(
+    client: Client,
+    start: str,
+    end: str,
+    limit: int,
+    ids_fetched_by_type: list[str],
+    event_type: EventType,
+    hide_sensitive: bool,
+) -> tuple[list[dict], Deduplicate]:
+    """
+    Fetch the event logs by type.
+    The fetch loop continues as long as it does not encounter one of the following:
+        - The size of the event list is equal to the limit
+        - Calling the api returned No content
+        - The nextToken did not return from the api call
+
+    Args:
+        client (Client): The client for api calls.
+        start (str), end (str): Start and end time period to retrieve logs.
+        limit (int): Maximum number of log items to return.
+        token (str | None): Token for retrieve the next set of log items.
+        ids_fetched_by_type (set[str] | None): ****
+        event_type (EventType): The type of event log to return.
+        hide_sensitive (bool): ****
+
+    Returns:
+        tuple[list[dict], Deduplicate]: List of the logs returned from trend micro, Deduplicate obj
+    """
+
+    params = assign_params(
+        start=start,
+        end=end,
+        type=event_type.value,
+    )
+
+    deduplicate_management = Deduplicate(ids_fetched_by_type, event_type)
+    next_token: str | None = None
+    events_res: list[dict] = []
+    while len(events_res) < limit:
+        params["limit"] = min(limit - len(events_res), 500)
+
+        try:
+            res = client.get_logs(event_type, params)
+        except NoContentException:
+            next_token = None
+            demisto.debug(
+                f"No content returned from api, {start=}, {end=}, {event_type.value=}"
+            )
+            break
+
+        if res.get("logs"):
+            # Iterate over each event log, update their `type` and `_time` fields
+            for event in res.get("logs"):
+                deduplicate_management.update_suspected_duplicate_events_list(
+                    event, start
+                )
+                if not deduplicate_management.is_duplicate(event, start):
+                    event.update(
+                        {"_time": event.get("timestamp"), "logType": event_type.value}
+                    )
+                    if hide_sensitive:
+                        remove_sensitive_from_events(event)
+
+                    add_missing_fields_to_event(event)
+                    events_res.append(event)
+
+        else:  # no logs
+            next_token = None
+            break
+
+        if next_token := res.get("nextToken"):
+            params["token"] = urllib.parse.unquote(next_token)
+        else:
+            next_token = None
+            demisto.debug(f"No `nextToken` for {event_type.value=}")
+            break
+
+    demisto.debug(f"Done fetching {event_type.value=}, got {len(events_res)} events")
```

Comment:
does `events_res` contain duplicates, or only de-duped items?

Reply:
It contains only de-duped, in line 313 it is filtered